src/execution/values.js

```diff
@@ -148,10 +148,9 @@ function coerceValue(type: GraphQLInputType, value: any): any {
     return Object.keys(fields).reduce((obj, fieldName) => {
       var field = fields[fieldName];
       var fieldValue = coerceValue(field.type, value[fieldName]);
-      if (isNullish(fieldValue)) {
+      if (fieldValue === undefined) {
         fieldValue = field.defaultValue;
-      }
-      if (!isNullish(fieldValue)) {
+      } else {
```

Comment:
this seems like a mistake - did your unit tests pass or are we missing a test for this condition? As I read it, if a fieldValue is coerced to be undefined, then we set `fieldValue` to the default value, however never actually set it on the response object.